libvcx/src/credential_def.rs

```diff
@@ -168,17 +166,22 @@ impl CredentialDef {
 
     fn get_state(&self) -> u32 { self.state as u32 }
 
-    fn rotate_rev_reg(&mut self) -> VcxResult<RevocationRegistry> {
+    fn rotate_rev_reg(&mut self, revocation_details: &str) -> VcxResult<RevocationRegistry> {
         debug!("rotate_rev_reg >>>");
-        let (tails_file, max_creds, issuer_did) = (self.get_tails_file(), self.get_max_creds(), self.issuer_did.as_ref());
+        let revocation_details = _parse_revocation_details(revocation_details)?;
+        let (tails_url, tails_file, max_creds, issuer_did) = (
+            revocation_details.tails_url.ok_or(VcxError::from_msg(VcxErrorKind::InvalidRevocationDetails, "tails_url not found in revocation details"))?,
+            revocation_details.tails_file.or(self.get_tails_file()),
```

Comment:
Not related to this PR, but `tails_file` is really confusing name when actually it's supposed to be a directory. I think we should make breaking change in some future PR and just rename it. What do you think?

Reply:
I agree with it being a confusing name, and it probably is worth making a breaking change

Reply:
I agree with it being a confusing name, and it probably is worth making a breaking change